@amifun/gccode.m

```diff
@@ -83,158 +83,90 @@
-            elseif(strcmp(this.cvar,'dJydx'))
-                cstr = regexprep(cstr, [this.cvar '\[([0-9+*]*)\] = '], [ this.cvar '[it+($1)*udata->nt] = ']);
-            elseif(strcmp(this.cvar,'dJzdx'))
-                cstr = regexprep(cstr, [this.cvar '\[([0-9+*]*)\] = '], [ this.cvar '[tdata->nroots[ie]+($1)*udata->nmaxevent] = ']);
-            end
-            
-            if(strfind(this.cvar,'Jy'))
-                cstr = regexprep(cstr,'my_([0-9]+)','my\[it+udata->nt*$1]');            
-                cstr = regexprep(cstr,'sigma_y_([0-9]+)','sigma_y\[$1\]');
-                cstr = regexprep(cstr,'dsdydp\[([0-9]*)\]','dsigma_ydp\[$1\]');
-                if(strcmp(this.cvar,'sJy'))
-                    cstr = regexprep(cstr,'sy_([0-9]+)',['sy\[it+udata->nt*$1]']);
-                    cstr = regexprep(cstr,'dJydy_([0-9]+)','dJydy\[$1\]');
-                    cstr = regexprep(cstr,'sJy\[([0-9]+)\+0\*([0-9]+)\]',['sJy\[$1\]']);
-                    cstr = regexprep(cstr,'sJy\[([0-9]+)\+([0-9]+)\*([0-9]+)\]',['s2Jy\[(($2-1)*$3+$1)\]']);
-                    cstr = strrep(cstr,'=','-=');
-                elseif(strcmp(this.cvar,'dJydy'))
-                    cstr = regexprep(cstr,'y_([0-9]+)','y\[it+udata->nt*$1\]');
-                    cstr = regexprep(cstr,'dJydy\[([0-9]+)\]\[([0-9]+)\]',['dJydy\[$1+' num2str(model.nytrue) '*$2\]']);
-                    cstr = regexprep(cstr,'dJydy\[([0-9]+)\+([0-9]+)\*([0-9]+)\]',['dJydy\[iy+($2*$3+$1)*' num2str(model.nytrue) '\]']);
-                elseif(strcmp(this.cvar,'dJydp'))
-                    cstr = regexprep(cstr,'y_([0-9]+)','y\[it+udata->nt*$1\]');
-                    cstr = regexprep(cstr,'dJydp\[([0-9]+)\+([0-9]+)\*([0-9]+)\]',['dJydp\[iy+($2*$3+$1)*' num2str(model.nytrue) '\]']);
-                else
-                    cstr = regexprep(cstr,'sy_([0-9]+)',['sy\[it+udata->nt*\($1+ip*' num2str(model.ny) '\)\]']);
-                    cstr = regexprep(cstr,'y_([0-9]+)','y\[it+udata->nt*$1\]');
-                    cstr = strrep(cstr,'=','+=');
-                end
-                cstr = regexprep(cstr,'dydp_([0-9]+)',['dydp\[$1+ip*' num2str(model.ny) ']']);
-                cstr = regexprep(cstr,'dydx_([0-9]+)','dydx\[$1]');
-            end
+            cstr = regexprep(cstr,'var_x_([0-9]+)','x_tmp[$1]');
+            cstr = regexprep(cstr,'var_sx_([0-9]+)','sx_tmp[$1]');
+            % sxdot needs to preces xdot
+            cstr = regexprep(cstr,'var_sxdot_([0-9]+)','sxdot_tmp[$1]');
+            cstr = regexprep(cstr,'var_xdot_([0-9]+)','xdot_tmp\[$1\]');
+            cstr = regexprep(cstr,'var_xBdot_([0-9]+)','xBdot_tmp\[$1\]');
+            cstr = regexprep(cstr,'xdot_old_([0-9]+)','xdot_old_tmp[$1]');
+            cstr = regexprep(cstr,'xdot_([0-9]+)','xdot_tmp[$1]');
+            cstr = regexprep(cstr,'var_xB_([0-9]+)','xB_tmp[$1]');
+            cstr = regexprep(cstr,'var_qBdot_([0-9]+)','qBdot_tmp\[ip + udata->nplist*$1]');
```

Comment:
you do not need to escape the escape string and actually easier to read if you don't. So I will remove unescessary escapes.